HesoyamHospital/WebApplication/Appointments/AppointmentController.cs

```diff
@@ -22,5 +26,15 @@ public IActionResult GetAllByPatient(long id)
             if (id != defaultPatientId) return BadRequest();
             return Ok(AppointmentMapper.AppointmentToAppointmentForObservationDto(_appointmentService.GetAllByPatient(id).ToList()));
         }
+
+        [HttpPut("cancel")]
+        public IActionResult Cancel([FromBody]long id)
+        {
+            Appointment appointment = _appointmentService.GetByID(id);
+            if (appointment == null) return NotFound();
+            if (!_appointmentValidation.IsPossibleToCancelAppointment(appointment, defaultPatientId)) return BadRequest();
```

Comment:
These kind of checks should not be on controller side.
Controller's only job is to return the proper status code and data.
Using Gergely's suggestion would be great: https://github.com/PSW-2020-ORG2/Hesoyam-Hospital/pull/58/files#r537099704